src/main/java/rx/schedulers/ExecutorScheduler.java

```diff
@@ -73,11 +66,15 @@ public Subscription schedule(Action0 action) {
             if (isUnsubscribed()) {
                 return Subscriptions.unsubscribed();
             }
-            ExecutorAction ea = new ExecutorAction(action, tasks);
+            ScheduledAction ea = new ScheduledAction(action, tasks);
```

Comment:
The problem with that is that remove from queue is O(n) and we'd pay it most of the time (once dequeued, remove will traverse the entire queue and not find it) because we can't distinguish between the normal completion calling unsubscribe and external unsubscribe.